applications/spring-shell/src/main/java/org/springframework/sbm/shell/ApplyShellCommand.java

```diff
@@ -80,3 +85,19 @@ public Availability availabilityCheck() {
         }
     }
 }
+
+@Component
+@RequiredArgsConstructor
+class ApplyRecipeValueProvider implements ValueProvider{
+
+    private final ProjectContextHolder projectContextHolder;
+    private final ApplicableRecipeListCommand applicableRecipeListCommand;
+
+    @Override
+    public List<CompletionProposal> complete(CompletionContext completionContext) {
+        ProjectContext projectContext = projectContextHolder.getProjectContext();
+        List<Recipe> applicableRecipes = applicableRecipeListCommand.execute(projectContext);
+        return applicableRecipes.stream().map(Recipe::getName).map(CompletionProposal::new).toList();
```

Comment:
Figured this one is so simple and duplicates parts of `ApplyShellCommand.apply`, that it does not warrant it's own unit test.